core/templates/components/question-directives/question-player/question-player-concept-card-modal.controller.ts

```diff
@@ -48,7 +48,15 @@ angular.module('oppia').controller('QuestionPlayerConceptCardModalController', [
     };
 
     $scope.retryTest = function() {
-      $window.location.replace(UrlService.getPathname());
+      $scope.selectedSubtopics = function() {
```

Comment:
Why is this attached to the scope? Also, why does it have to be a function if it is only used once and just here?

Can we use the logic inline to populate subtopicIds and then use that value to set the URL?

Reply:
done.